src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -637,7 +694,11 @@ private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
       bldr.setSslcontext(sslContext);
     }
 
-    HttpClient httpClient = bldr.build();
+    HttpClient httpClient;
+    // this lets the user perform non-essential configuration (e.g. timeouts)
+    // but reduces the risk of essentials not being set. MK.
+    HttpClientBuilder b = configurator.configure(bldr);
```

Comment:
I can do that but we explicitly reject `null` configurator values at the moment and it makes more sense to me that way.